### PR TITLE
Bump baseline, POM, BOM, etc.

### PR DESCRIPTION
**Pull request by jglick**
Supersedes #352, #351, & #314.

### CODE REVIEW

File: .jenkins/artifacts.lst
Comment (jglick) [inline .jenkins/artifacts.lst:1]: who knows

**Comment by MarkEWaite** [comment]
With a minor update to use parent pom 4.59 and bom version 1981.v17df70e84a_a_1 this would also supersede #356 and #360.

I see one test failure 

```
[ERROR] jenkins.branch.BranchCategoryFilterTest.filter_uncategorizedCategory  Time elapsed: 0.039 s  <<< FAILURE!
java.lang.AssertionError:

Expected: iterable with items [<Mock for TopLevelJob, hashCode: 1652280702>, <Mock for TopLevelJob, hashCode: 889546737>] in any order
     but: no match for: <Mock for TopLevelJob, hashCode: 442200980>
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
        at jenkins.branch.BranchCategoryFilterTest.filter_uncategorizedCategory(BranchCategoryFilterTest.java:94)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
```

That failure is visible on Debian testing and on Ubuntu 22.04.

**Comment by jglick** [comment]
> this would also supersede

One step at a time. I wanted to file a PR which covered the transition to Java 11. Individual updates can be applied on their own merits. (Some, like @jtnord, object to applying `bom` updates indiscriminately.)

**Comment by MarkEWaite** [comment]
> If you mean after applying _further_ BOM updates, then let us address that there.

After further investigation, it looks like the test failure requires a newer parent pom (4.59) but does not require any change to the BOM version.  This passes with parent pom 4.53 as proposed:

`mvn clean -Dtest=BranchCategoryFilterTest#filter_uncategorizedCategory clean verify`

This fails with parent pom 4.59 as the only change from this pull request

`mvn clean -Dtest=BranchCategoryFilterTest#filter_uncategorizedCategory clean verify`

Handling that test failure can certainly wait until after this pull request is merged.  Updating from parent pom 4.50 to 4.53 is already a step in the right direction.

**Review by jtnord — APPROVED** [review]

**Pull request merged**
